### DISCUSSION TITLE
Repeaters randomly not working

### DISCUSSION THREAD

**Discussion by diedder** [Budibase/budibase#6418] [Help]
Hi,

my first project is going on, I put some first real data in the DB and found some strange behavior. The structure is as follows:
- Data Provider SQL Query with Questions
-- Repeater for the Questions, displayed in "self-made  cards"
---in each "card" three parallel Data Providers with SQL wich queries special relations of the single question in the card
---- also three Repeater (one below each data provier) to show the relations as "self-made tags"
overall there are a lot of data providers / repeaters on this page. Im my case 10 question = 10 + 10*3 = 40. The repeaters in the cards are not working reliable. Everytime I open the page not all repeaters are showing the realtion-data, some stay empty. But the empty ones are changing, see attached screenshots (always the same data!).

My question: is there a limitation for the data provider / repeater count when they are cascading? The page itself loads fast, so a timeout should not be the issue. Is there a log I can inspect for further investigations?

![image](https://user-images.githubusercontent.com/25912001/174961456-544c6592-e56a-4547-99ba-16428e7de613.png)
![image](https://user-images.githubusercontent.com/25912001/174961511-4de32357-c2a4-4bd1-811d-65984696a5b8.png)
![image](https://user-images.githubusercontent.com/25912001/174961565-389090d4-c7c3-495d-960f-950166054209.png)

br, diedder

**Comment by diedder** [comment]
update: I reduced the number of data providers with SQl query. There is now only one SQL query per card and the three data providers / repeaters are feeded of this. Effect: all repeaters are working fine, no data missing.

maybe there are too much SQL queries? A log would be very interesting. I'll try to check the log from the SQLserver also.

**Reply by melohagan** [reply]
I would also verify that all of the Questions would definitely return relationships with your SQL.
There very well may be a data limit to prevent the designer/app from crashing/freezing. 

Are you self-hosted?

**Comment by diedder** [comment]
yes, they should display data. The issue of randomly missing data occurs although the query should return 1-3 rows. Also interesting, the missing data is not always at the same question.

I checked the SQL server log. The log validates my suspect. There should be 10x3=30 select queries for the relations, but there are only 21 in the log. The number of this query is also not constant, but there are always some queries missing / not sent to the server by budibase.

No, I'm using the cloudservice for now. I'll change to selfhosting late. btw are there recommandation for CPU/RAM for a vserver?

**Reply by melohagan** [reply]
It would be helpful if you could provide us the app export and a SQL init script to support@budibase.com
Then I can have a go at replicating the issue and investigating further.

**Comment by diedder** [comment]
Hi @melohagan ,
sure, I'll prepare that for you. What do you mean by "init script"? The credentials for the external SQL server?

**Reply by melohagan** [reply]
A script that would allow me to reproduce the problem. So the Create Table scripts and some insert statements for test data. 
Credentials for an external SQL Server to do the read could be handy instead if its just test data?

**Reply by melohagan** [reply]
Thanks @diedder 

I had a look at the app export. It does appear that not all of the Questions do have tags associated.
For example ID = 1 for query is good
![Screenshot 2022-06-23 at 17 11 22](https://user-images.githubusercontent.com/101575380/175345579-1764e2de-9ae7-4bd4-a27c-e78c015ab552.png)
 
But ID = 2 has no results:
![Screenshot 2022-06-23 at 17 12 24](https://user-images.githubusercontent.com/101575380/175345817-9f4669e1-6899-481c-816e-19e515de84d5.png)

Also your filters look wrong to me:
![Screenshot 2022-06-23 at 17 14 46](https://user-images.githubusercontent.com/101575380/175346211-f1e46b17-30ea-4140-a06a-513f1350d075.png)
![Screenshot 2022-06-23 at 17 15 07](https://user-images.githubusercontent.com/101575380/175346264-18610521-8adb-42e7-ab71-7e16212c47bd.png)

**Reply by diedder** [reply]
Hi @melohagan 

yes, correct. Not all question have relationsships. Here ID=2 is the only one. but this doesn't matter - see the working site.
Also the filtering is on the working site, pls see the not working (red backgorund) site for your investigations.

**Reply by diedder** [reply]
Hi @melohagan,
any news at this topic? I have a similar page which has the same issue. If there are some (app. > 15) lines from a repeater the sub-repeaters in the lines are randomly not working, so the displayed queryresult is not reliable... The page is loading fast, so there should no timeout for the queries.
Thanks!

**Comment by melohagan** [comment]
Hi @diedder 

Apologies for not getting back sooner. I've had another look there and I can't spot the difference!

| RED | GREEN |
| --- | --- |
| ![Screenshot 2022-07-01 at 17 18 32](https://user-images.githubusercontent.com/101575380/176932561-2fa365ec-61a8-45a2-b91e-53dbbf272974.png) | ![Screenshot 2022-07-01 at 17 18 50](https://user-images.githubusercontent.com/101575380/176932617-2218651f-e79e-4f27-ad4d-fbdd50c5cf38.png) |

Hopefully I've understood correctly that you expect the blue tags to differ in number between the screens?
I'm on app version **1.0.214**

Although you said in your email that the number of repeaters per card would be different in each screen but they also look the same between `questions/list/:standard` and `questions2/list/:standard`

| RED | GREEN |
| --- | --- |
| ![Screenshot 2022-07-01 at 17 24 21](https://user-images.githubusercontent.com/101575380/176933562-387888b7-4bbf-43e8-8b73-daa923150a35.png) | ![Screenshot 2022-07-01 at 17 24 54](https://user-images.githubusercontent.com/101575380/176933635-1754d9e9-cdf6-4323-9862-8887f62918b8.png) |

I do see an extra data provider in the green one: *Data Provider Tags*


----

Are you self-hosted? Could it possibly be a difference in machine resources? Here's my docker resources:
<img width="937" alt="Screenshot 2022-07-01 at 17 41 05" src="https://user-images.githubusercontent.com/101575380/176935953-65a7433e-fcb3-4001-b4ce-5e032cefc704.png">

That's probably overkill.

**Comment by diedder** [comment]
Hi @melohagan 
no need for excuses, I am very grateful for your help!
Very strange that there is no difference at your system... I checked again an used the same app I sent you, also on 1.0.214 an the red/green pages provide different results. Here a screenshot with both pages directly side by side
<img width="581" alt="Bildschirmfoto 2022-07-01 um 23 58 20" src="https://user-images.githubusercontent.com/25912001/176992306-6beb1212-3dfd-4a30-baeb-d068ce0d3eeb.png">
<img width="565" alt="Bildschirmfoto 2022-07-01 um 23 57 59" src="https://user-images.githubusercontent.com/25912001/176992303-0d0e1e59-4561-4cb3-ac62-62284cf15b6c.png">
In total 5 tags (blue boxes) are missing!

Clarification of my statement to the repeater. The red/green screens differ in the total number of SQL queries made. In the green one the additional data provider does one query and all other data providers / repeaters reference to this one. In the red page every tag does it's own SQL query in one data provider. I inserted the additional data provider only for the quick investigations, it should not be the final design.

I'm using the cloud service, not self hosted. I tried to check with a local docker installations but the compose fails. The page is not reachable due to not all images get running, some are continuously restarting... 
<img width="672" alt="Bildschirmfoto 2022-07-02 um 10 11 22" src="https://user-images.githubusercontent.com/25912001/176992654-c5d2c622-64e0-4707-bb7b-c32bedbceca4.png">
Sorry, i cannot check if the issue depends on the hosting type for now.

Have a nice weekend!

**Comment by melohagan** [comment]
Hi @diedder 

Thanks for the additional information! I should have asked if you were cloud hosting! What is your tenant id?

This your subdomain e.g. `<tenantid>.budibase.app` 
It may be possible that you're running into some query limit in the cloud.

**Reply by diedder** [reply]
Hi @melohagan 
My tenant is `bqrc` In my account are currently two apps active, my live app (pls do not touch) and the one I sent you. I CAN also invite you as user to this.

**Comment by diedder** [comment]
I have tried the project in self hosted docker environment an can confirm that this must be an limitation / issue of the cloud hosting. In self hosted version everything works fine.

**Comment by diedder** [comment]
Hi @melohagan 
I want to reopen this topic because I have some news. Not the self-hosting or cloud installation rather then the SSL ecnryption via a reverse proxy makes the difference!
In my local docker dekstop installation everthing worked fine so I booked a VPS an installed budibase working with nginx as reverse proxy. The VPS ressource are big enough. With SSL enabled in the proxy the repeater-issue is still there, turnig SSL off everything works perfect. In the acces/error logs I could find any useful information about it.

could you try to reproduce this?
